src/components/metrics.tsx

```diff
@@ -436,6 +439,30 @@ export const QueryTable: React.FC<QueryTableProps> = ({ index, namespace }) => {
     }
   }
 
+  const getCsvColumns = () => {
+    const csvColumns = [];
+    for (let i = 1; i < columns.length; i++) {
+      const column = columns[i];
+      if (typeof column?.title === 'string') {
+        csvColumns.push(column.title);
+      } else if (column.title?.props?.children) {
+        csvColumns.push(column.title.props.children);
+      }
+    }
+    return csvColumns;
+  };
+
+  const getCsvRows = () => {
+    const csvRows = [];
+    for (let i = 0; i < rows.length; i++) {
+      const row = [...rows[i]].slice(1);
+      csvRows.push(row);
+    }
+    return csvRows;
+  };
+
+  const csvData = [getCsvColumns(), ...getCsvRows()];
```

Comment:
This is doing the work of preparing the CSV data for download every time the table is rendered. It would be much more efficient if we only generated the CSV data when the download button is clicked and it looks like `react-csv` does support that approach. (We've had trouble with performance on this Metrics page in the past and have worked hard to make it more efficient.)

Reply:
Good advice, makes sense! I updated this in the latest commit so that the CSV is only generated `onClick` when the user selects 'Export as CSV' from the Kebab menu.